sbi/simulators/nonlinear_gaussian.py

```diff
@@ -112,104 +111,99 @@ def get_ground_truth_posterior_samples_nonlinear_gaussian(
+        m0, m1, s0, s1, r = self._unpack_params(theta)
 
-        us = np.random.randn(num_simulations, self._num_observations_per_parameter, 2)
-        observations = np.empty_like(us)
+        us = np.random.randn(num_simulations, self._num_xs_per_parameter, 2)
+        xs = np.empty_like(us)
 
-        observations[:, :, 0] = s0 * us[:, :, 0] + m0
-        observations[:, :, 1] = (
-            s1 * (r * us[:, :, 0] + np.sqrt(1.0 - r ** 2) * us[:, :, 1]) + m1
-        )
+        xs[:, :, 0] = s0 * us[:, :, 0] + m0
+        xs[:, :, 1] = s1 * (r * us[:, :, 0] + np.sqrt(1.0 - r ** 2) * us[:, :, 1]) + m1
 
         mean, std = self._get_observation_normalization_parameters()
         return (
-            torch.Tensor(
-                observations.reshape(
-                    [num_simulations, 2 * self._num_observations_per_parameter]
-                )
-            )
+            torch.Tensor(xs.reshape([num_simulations, 2 * self._num_xs_per_parameter]))
             - mean.reshape(1, -1)
         ) / std.reshape(1, -1)
 
-    def log_prob(self, observations, parameters):
-        """Log likelihood of observations given parameters. 
+    def log_prob(self, x: torch.Tensor, theta: torch.Tensor):
+        """Log likelihood of observations given parameter sets theta.
         
-        Likelihood is proportional to a product of self._num_observations_per_parameter 2D
-        Gaussians and so log likelihood can be computed analytically.
+        Likelihood is proportional to a product of self._num_xs_per_parameter
+         2D Gaussians and so log likelihood can be computed analytically.
         
         Arguments:
-            observations {torch.Tensor} [batch_size, observation_dim] -- Batch of observations.
-            parameters {torch.Tensor} [batch_size, parameter_dim] -- Batch of parameters.
+            x: Batch of observations of shape (batch_size, observation_dim).
+            theta: Batch of parameters (batch_size, parameter_dim).
         
         Returns:
-            torch.Tensor [batch_size] -- [Log likelihood log p(x | theta) for each item in the batch.
+            torch.Tensor [batch_size] -- [Log likelihood log p(x | theta) for each item
+             in the batch.
         """
 
-        if isinstance(parameters, torch.Tensor):
-            parameters = utils.tensor2numpy(parameters)
+        if isinstance(theta, torch.Tensor):
+            theta = utils.tensor2numpy(theta)
 
-        if isinstance(observations, torch.Tensor):
-            observations = utils.tensor2numpy(observations)
+        if isinstance(x, torch.Tensor):
+            x = utils.tensor2numpy(x)
 
-        if observations.ndim == 1 and parameters.ndim == 1:
-            observations, parameters = (
-                observations.reshape(1, -1),
-                parameters.reshape(1, -1),
+        if x.ndim == 1 and theta.ndim == 1:
+            x, theta = (
+                x.reshape(1, -1),
```

Comment:
@janfb, this still needed?

Reply:
`x` is 2D for sure. `theta` could be 1D if we sample from the prior via `prior.sample()` instead of `prior.sample((1, ))`.